mage/query-modules/cpp/pagerank_online.md

```diff
@@ -0,0 +1,220 @@
+[PageRank](pagerank.md) in a graph streaming scenario. Incremental- local
+changes are introduced in the algorithm to prevent users from recalculating
+PageRank values each time a change occurs in the graph (something is added or
+deleted).
+
+To make it as fast as possible, the online algorithm is only the approximation
+of PageRank but carrying the same information - the likelihood of random walk
+ending in a particular vertex.  The work is based on "[Fast Incremental and
+Personalized
+PageRank](http://snap.stanford.edu/class/cs224w-readings/bahmani10pagerank.pdf)"
+[^1], where authors are deeply focused on providing the streaming experience of
+a highly popular graph algorithm.
+
+Approximating PageRank is done simply by exploring random walks and calculating
+the frequency of a node within all walks. `R` walks are sampled by using a
+random walk with a stopping probability of `eps`.Therefore, on average, walks
+would have a length of `1/eps`. Approximative PageRank is based on the formula
+below:
+
+```
+RankApprox(v) = X_v / (n * R / eps)
+```
+
+Where `X_v` is the number of walks where the node `v` appears.  The theorem
+written in the paper explains that RankApprox(v) is sharply concentrated around
+its expectation, which is Rank(v).
+
+### Usage
+
+Online PageRank should be used in a specific way. To set the parameters, the
+user should call a `set()` procedure. This function also sets the context of a
+streaming algorithm. `get()` function only returns the resulting values stored
+in a cache. Therefore, if you try to get values before first calling `set()`,
+the run will fail with a proper message.
+
+To make the incremental flow, you should set the proper trigger. For that, we'll
+use the `update()` function:
+
+```cypher
+CREATE TRIGGER pagerank_trigger
+(BEFORE | AFTER) COMMIT
+EXECUTE CALL pagerank_online.update(createdVertices, createdEdges, deletedVertices, deletedEdges) YIELD *
+SET node.rank = rank;
+```
+
+Finally, the `reset()` function resets the context and enables the user to start
+new runs.
+
+[^1] [Fast Incremental and Personalized
+PageRank](http://snap.stanford.edu/class/cs224w-readings/bahmani10pagerank.pdf),
+Bahman Bahmani et al.
+
+| Trait               | Value                                                 |
+| ------------------- | ----------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**algorithm**</Highlight>  |
+| **Implementation**  | <Highlight color="#FB6E00">**C++**</Highlight>        |
+| **Graph direction** | <Highlight color="#FB6E00">**directed**</Highlight>   |
+| **Edge weights**    | <Highlight color="#FB6E00">**unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight> |
+
+## Procedures
+
+### `set(walks_per_node, walk_stop_epsilon)`
+
+#### Input:
+
+* `walks_per_node: int(10)` ➡ Number of sampled walks per node.
+* `walk_stop_epsilon: double(0.1)` ➡ The probability of stopping when deriving
```

Comment:
I removed a sentence fragment from here: "On average it will create".
Please check if it was intentionally placed here @jmatak.

Reply:
`On average it will create walks of length 1 / walk_stop_epsilon`

This is what it should have been written. Include it if you think it is useful :smile: